src/Microsoft.Extensions.Caching.Memory/CacheEntry.cs

```diff
@@ -9,7 +9,7 @@
 
 namespace Microsoft.Extensions.Caching.Memory
 {
-    internal class CacheEntry : ICacheEntry
+    public class CacheEntry : ICacheEntry
```

Comment:
CacheEntry is the wrong view to expose to the eviction algorithm. This view is the one used for creating the entry so Value is settable, all the expirations are settable, AbsoluteExpirationRelativeToNow is relative to entry creation rather than the expiration scan, etc..

There should be a different interface/view exposed for creation vs eviction, even if the same object implements both internally.